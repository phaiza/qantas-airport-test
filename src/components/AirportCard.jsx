import { Link } from 'react-router-dom';
import styles from './AirportCard.module.scss';

const AirportCard = ({ airport }) => {
  const { airportName, city, country, airportCode } = airport;

  return (
    <Link
      to={`/airport/${airportCode}`}
      className={styles.airportCard}
      aria-label={`View airport details for ${airportName}, located in ${city?.cityName}, ${country?.countryName}`}
    >
      <div className={styles.airportCard__header}>
        <div className={styles.airportCard__text}>
          <h2 className={styles.airportCard__name} role="heading">
            {airportName}
          </h2>
          <span className={styles.airportCard__code}>({airportCode})</span>
        </div>
        <p className={styles.airportCard__meta}>
          {city?.cityName}, {country?.countryName}
        </p>
      </div>
      <div className={styles.airportCard__chevron}>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 24 24"
          fill="none"
          stroke="currentColor"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          aria-hidden="true"
          focusable="false"
        >
          <polyline points="9 18 15 12 9 6" />
        </svg>
      </div>
    </Link>
  );
};

export default AirportCard;
