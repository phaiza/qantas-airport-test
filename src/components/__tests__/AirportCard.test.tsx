import '@testing-library/jest-dom/vitest';

import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { describe, it, expect } from 'vitest';
import AirportCard from '../AirportCard';

const mockAirport = {
  airportCode: 'AAA',
  airportName: 'Anaa',
  city: {
    cityName: 'Anaa',
  },
  country: {
    countryName: 'French Polynesia',
  },
};

describe('AirportCard', () => {
  it('renders airport name, code, and location', () => {
    render(
      <BrowserRouter>
        <AirportCard airport={mockAirport} />
      </BrowserRouter>
    );

    expect(screen.getByText('Anaa')).toBeInTheDocument(); // name
    expect(screen.getByText('(AAA)')).toBeInTheDocument(); // code
    expect(screen.getByText('Anaa, French Polynesia')).toBeInTheDocument(); // location
  });

  it('navigates to correct detail page', () => {
    render(
      <BrowserRouter>
        <AirportCard airport={mockAirport} />
      </BrowserRouter>
    );

    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('href', '/airport/AAA');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <AirportCard airport={mockAirport} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
