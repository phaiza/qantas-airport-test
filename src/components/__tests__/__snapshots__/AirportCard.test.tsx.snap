// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AirportCard > matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="_airportCard_b97b4f"
    data-discover="true"
    href="/airport/AAA"
  >
    <div
      class="_airportCard__header_b97b4f"
    >
      <div
        class="_airportCard__text_b97b4f"
      >
        <h3
          class="_airportCard__name_b97b4f"
        >
          Anaa
        </h3>
        <span
          class="_airportCard__code_b97b4f"
        >
          (AAA)
        </span>
      </div>
      <p
        class="_airportCard__meta_b97b4f"
      >
        Anaa, French Polynesia
      </p>
    </div>
    <div
      class="_airportCard__chevron_b97b4f"
    >
      <svg
        aria-hidden="true"
        fill="none"
        focusable="false"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
    </div>
  </a>
</DocumentFragment>
`;
